tensorflow_datasets/image/caltech10k_webfaces.py

```diff
@@ -0,0 +1,116 @@
+
+Pruning Training Sets for Learning of Object Categories
+Anelia Angelova, Yaser Abu-Mostafa, Pietro Perona
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import tensorflow as tf
+
+import tensorflow_datasets.public_api as tfds
+
+LANDMARK_HEADINGS = ("lefteye_x lefteye_y righteye_x righteye_y "
+           "nose_x nose_y mouth_x mouth_y").split()
+IMAGE_DATA = ("http://www.vision.caltech.edu/Image_Datasets/Caltech_10K_WebFaces/Caltech_WebFaces.tar")
+LANDMARK_DATA = ("http://www.vision.caltech.edu/Image_Datasets/Caltech_10K_WebFaces/WebFaces_GroundThruth.txt")
+
+_CITATION = """\
+@article{Angelova05learningobjCategories,
+  title={Pruning training sets for learning of object categories},
+  author={ A. Angelova and Y. Abu-Mostafam and P. Perona },
+  journal={Conference on Computer Vision and Pattern Recognition},
+  year={2005},
+}
+"""
+_DESCRIPTION = """\
+The dataset contains images of people collected from the web by typing common given names into 
+Google Image Search. The coordinates of the eyes, the nose and the center of the mouth for each 
+frontal face are provided in a ground truth file. This information can be used to align and crop 
+the human faces or as a ground truth for a face detection algorithm. The dataset has 10,524 human 
+faces of various resolutions and in different settings, e.g. portrait images, groups of people, etc. 
+Profile faces or very low resolution faces are not labeled.
+""" 
+class Caltech10K_WebFaces(tfds.core.GeneratorBasedBuilder):
+
+  VERSION = tfds.core.Version("0.1.0")
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+      builder=self,
+      description=_DESCRIPTION,
+      features=tfds.features.FeaturesDict({
+        "image": tfds.features.Image(encoding_format='jpeg'),
+        "landmarks": {name: tf.int64 for name in LANDMARK_HEADINGS}
+      }),
+      urls=['http://www.vision.caltech.edu/Image_Datasets/Caltech_10K_WebFaces/'],
+      citation=_CITATION
+    )
+
+
+  def _split_generator(self, dl_manager):
+    extracted_dirs = dl_manager.download_and_extract({
+      "images": IMAGE_DATA,
+      "landmarks": LANDMARK_DATA
+    })
+    return [
+      tfds.core.SplitGenerator(
+        name=tfds.Split.TRAIN,
+            num_shards=5,
+            gen_kwargs={
+                "extracted_dirs": extracted_dirs
+            })
+    ]
+
+
+  def _process_caltech10k_config_file(self, file_path):
+    with tf.io.gfile.GFile(file_path) as f:
+      data_raw = f.read()
+    lines = data_raw.split("\n")
+
+    keys = [x.split()[0] for x in lines[:-1]]
+    values = {}
+    # Go over each line (skip the last one, as it is empty).
+    for line in lines[0:-1]:
+      row_values = line.strip().split()
+      # Each row start with the 'file_name' and then space-separated values.
+      values[row_values[0]] = [v for v in row_values[1:]]
+    return keys, values
```

Comment:
please explain more what are the "keys" and "values" here..

keys should be the same as "LANDMARK_HEADINGS" right ?